### ISSUE TITLE
KB stops reporting values but is still on

### ISSUE THREAD

**Issue by lily-bit-0** [estokes/pinephone-kbpwrd#4]
Reoccurringly in all states (including while charging), the KB stops reporting any values as if it had been shut off via the button or automatically. kbpwrd also keeps saying "setting online" and "invalid cross-device link" in these cases. But its not actually turned off, the KB battery still supplies the PP with power. Shutting it off and on again using the button fixes it, at least until it happens again. Outside of kbpwrd I’ve never experienced the keyboard battery not reporting any values while its still on and supplying power, especially with a charger plugged in.

**Comment by lily-bit-0** [comment]
It looks like there is a chance of this happening whenever writing to `/sys/class/power_supply/ip5xxx-boost/online`. I’m attempting to write a Bash script to make automatic limits work for me and the same thing is happening. When I temporarily had the script write `1` to `online` every second, it happened very frequently, sometimes after a couple of seconds. As soon as I removed all lines involving `online` it stopped happening entirely. My systemd services that wrote to `online` to shut KB->PP charging off and on during suspend had a chance of two different undesired outcomes: Either this bug where no values are reported, KB powering still works and it’s fixed by using the button to shut it off and then on again, or it can entirely shut off as normal, what Megi calls the charger going to sleep, where I can turn it on again right away with one button press. Judging by your code comments, this can happen when `online` is set to 0 for some time. So it looks like there are two separate bugs involving `online`. And in both cases `status` reports `Not charging` instead of `Full`, `Charging` or `Discharging`. I didn’t know that before as I only started monitoring `status` after these bug reports when I started working on my own limit script, I guess I assumed `Not charging` was a normal state when I tested kbpwrd, which it can be when the KB is off. #3 also displays this status when broken.
